docs/aws/audit/redshiftmonitoring/rules/redshift_instance_generation_remediation.mdx

```diff
@@ -1,6 +1,223 @@
+
+4. **Update Node Type**:
+   - If the node type is not the latest generation, the script modifies the cluster to use the latest generation node type.
+
+This script will help you automate the process of ensuring that your Redshift clusters are always using the latest generation of nodes.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard by selecting "Redshift" under the "Database" category.
+3. In the navigation pane, click on "Clusters". This will display a list of all your Redshift clusters.
+4. For each cluster, check the "Node Type" column. If the node type is not the latest generation (for example, dc2.large), then it indicates that the Redshift cluster is not using the latest generation of nodes.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the Redshift resources.
+
+2. Once the AWS CLI is set up, you can list all the Redshift clusters using the following command:
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NodeType]'
+   ```
+   This command will return a list of all Redshift clusters along with their node types.
+
+3. To check if the Redshift clusters are using the latest generation of nodes, you need to compare the NodeType of each cluster with the latest generation node types. As of now, the latest generation node types are 'dc2' and 'ra3'.
+
+4. You can use a Python script to automate this process. The script will use the AWS SDK (boto3) to list all the Redshift clusters and their node types, and then compare them with the latest generation node types. Here is a simple example:
+
+   ```python
+   import boto3
+
+   # Create a Redshift client
+   redshift = boto3.client('redshift')
+
+   # List all Redshift clusters
+   clusters = redshift.describe_clusters()
+
+   # Check if the clusters are using the latest generation of nodes
+   for cluster in clusters['Clusters']:
+       if cluster['NodeType'] not in ['dc2', 'ra3']:
+           print(f"Cluster {cluster['ClusterIdentifier']} is not using the latest generation of nodes.")
+   ```
+   This script will print the identifiers of the clusters that are not using the latest generation of nodes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including Redshift.
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>